src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,169 @@
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+import { types } from "sass";
+import Null = types.Null;
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly approvedByRegistrar!: boolean
+
+  /** text used in radio options . */
+  @Prop({ default: false }) readonly isConversionToFullRestoration!: boolean
+
+  /** Show only the court order option; remove via registrar option. */
+  @Prop({ default: false }) readonly isCourtOrderOnly!: boolean
+
+  // Local properties
+  private courtOrderNumber = ''
+  private courtOrderNumRules = []
+  private valid = false
+  private approvalTypeSelected = ''
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // Copy props to mutable properties
+    if (this.approvedByRegistrar) {
+      this.courtOrderNumber = ''
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_REGISTRAR
+    }
+    if (this.draftCourtOrderNumber) {
+      this.courtOrderNumber = this.draftCourtOrderNumber
+      this.approvalTypeSelected = this.ApprovalTypes.VIA_COURT_ORDER
+    }
+  }
+
+  /** Triggers the form validation. */
+  public validate (): boolean {
+    let status = this.$refs.courtNumRef.validate()
+    this.$emit('emitValid', status)
+    return status
+  }
+
+  private emitValidationError (event): void {
+    this.$emit('emitValid', !event)
+  }
+
+  @Emit('emitRadioButtonChange')
+  private radioButtonChanged (event): void {
+    if (event === this.ApprovalTypes.VIA_REGISTRAR) {
+      Vue.set(this, 'courtOrderNumber', '')
+      Vue.set(this, 'courtOrderNumRules', [])
+    }
+    if (event === this.ApprovalTypes.VIA_COURT_ORDER) {
+      this.setCourtNumberRules()
+    }
+    this.validate()
+  }
+
+  @Emit('emitCourtNumberChange')
+  private courtOrderNumberChanged (event): void {
+    Vue.set(this, 'approvalTypeSelected', this.ApprovalTypes.VIA_COURT_ORDER)
+    this.$emit('emitRadioButtonChange', this.ApprovalTypes.VIA_COURT_ORDER)
+    this.setCourtNumberRules()
+    this.validate()
+  }
+
+  private setCourtNumberRules (): void {
+    // Apply TextField rules
+    this.courtOrderNumRules = [
+      (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
+      (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
+      (v: string) => (!v || !(v.length < 5)) || 'Court order number is invalid',
+      (v: string) => (!v || !(v.length > 20)) || 'Court order number is invalid',
+      (v: string) => !!v || 'A Court Order number is required'
+    ]
```

Comment:
This is fine, but FYI, another way this can be done is in the component itself, like this:
```
   :rules='enableValidation ? courtNumberRules : []
```

Where enableValidation is a flag you togle and courtNumberRules is a readonly const.

Reply:
Do these rules ever need to be cleared?

Reply:
Good to know.  I'll leave this as is for now, but I appreciate the feedback.

Reply:
Do these rules ever need to be cleared?

Reply:
No, not as far as I know.  Do I need a public clear() method?  Is that what you're suggesting?

Reply:
I was thinking that you might need to set the rules to [] to disable them.

But maybe this is "once enabled, always enabled".